docs/content/concepts/assets/external-assets.mdx

```diff
@@ -0,0 +1,166 @@
+
+- Data that is landed by an external source (e.g. a file landing daily in s3)
+- Data that is created and processed using manual processes
+- Data that is materialized by existing pipelines with their own scheduling and infrastructure that you do not want to or need to migrate en masse.
+
+With an external assets you can:
+
+- Attach metadata to its definition for documentation, tracking ownership, and so on
+- Track its data quality in Dagster.
+- Track its data version in Dagster
+- Schedule downstream assets based on updates to external assets using asset sensors or auto-materialize policies
+
+But you cannot
+
+- Schedule its materialiation
+- Backfill it using Dagster
+- Use the UI or GraphQL API to instigate ad hoc materializations.
+
+## Relevant APIs
+
+| Name                                             | Description                                                                             |
+| ------------------------------------------------ | --------------------------------------------------------------------------------------- |
+| <PyObject object="external_assets_from_specs" /> | Create a list of external asset definitions from a list of specs                        |
+| <PyObject object="AssetSpec" />                  | An object that represents the metadata of a particular asset                            |
+| <PyObject object="SensorResult" />               | Result object for a sensor evalation where you pass events to attach to external assets |
+| <PyObject object="DagsterInstance" />            | Object presenting the Dagster instance                                                  |
+
+## Defining External Assets
+
+### A single external asset
+
+The following code declares a single external asset and passes it to <PyObject object="Definitions"/>
+
+```python file=/concepts/assets/external_assets/single_declaration.py
+from dagster import AssetSpec, Definitions, external_asset_from_spec
+
+defs = Definitions(assets=[external_asset_from_spec(AssetSpec("my_asset"))])
+```
+
+### External assets with dependencies
+
+External assets can depend on other external assets. You do this by using the `deps` argument of <PyObject object="AssetSpec" />.
+
+```python file=/concepts/assets/external_assets/external_asset_deps.py
+from dagster import AssetSpec, Definitions, external_assets_from_specs
+
+asset_one = AssetSpec("asset_one")
+asset_two = AssetSpec("asset_two", deps=[asset_one])
+
+defs = Definitions(assets=external_assets_from_specs([asset_one, asset_two]))
+```
+
+## Keeping an external asset's metadata up to date.
+
+Since with external assets Dagster does not control scheduling or materialization, it is up to the user to keep the metadata for an external asset up-to-date. There are a number of mechanisms for doing so:
+
+- REST APIs
+- Sensors
+- Log events in ops
+
+### Using REST API
+
+TODO: joe
```

Comment:
```bash
curl --request POST \
  --url https://{organization_name}.dagster.cloud/{deployment_name}/report_asset_materialization/{asset_key}\
  --header 'Content-Type: application/json' \
  --header 'Dagster-Cloud-Api-Token: {user_token}' \
  --data '{
	"data_version": "gitsha",
	"partition": "2021-08-01",
	"description": "an asset materialized externally from dagster",
	"metadata" : {
		"some_key": 100,
		"another_key": "hi"
	}
}'
```


```python
import requests

url = f"https://{organization_name}.dagster.cloud/{deployment_name}/report_asset_materialization/{asset_key}"

payload = {
    "data_version": "testing",
    "partition": "",
    "description": "external asset",
    "metadata": {
        "some_key": 100,
        "another_key": "hi"
    }
}
headers = {
    "Content-Type": "application/json",
    "Dagster-Cloud-Api-Token": USER_TOKEN
}

response = requests.request("POST", url, json=payload, headers=headers)

print(response.text)
```

For OS you can drop the org_name, deployment_name and api token header

Reply:
> ```shell
> curl --request POST \
>   --url https://{organization_name}.dagster.cloud/{deployment_name}/report_asset_materialization/{asset_key}\
>   --header 'Content-Type: application/json' \
>   --header 'Dagster-Cloud-Api-Token: {user_token}' \
>   --data '{
> 	"data_version": "gitsha",
> 	"partition": "2021-08-01",
> 	"description": "an asset materialized externally from dagster",
> 	"metadata" : {
> 		"some_key": 100,
> 		"another_key": "hi"
> 	}
> }'
> ```

where does `asset_key` get specified here?

Reply:
@Ramshackle-Jamathon still looking for guidance here

Reply:
@schrockn sorry for the delay updated my initial comment